content/en/blog/2023/otel-dotnet-performance-impacts.md

```diff
@@ -0,0 +1,287 @@
+---
+title: Does OpenTelemetry in .NET Cause Performance Degradation?
+linkTitle: opentelemetry dotnet performance impacts
+date: 2023-03-10
+author: '[MartinDotNet](https://github.com/martinjt) (Honeycomb)'
+---
+
+Contrary to Betteridge’s Law of Tabloid Headlines, the answer is “yes,” but
+context is important. I get this question so often that I thought it was time to
+get some stats on it.
+
+I’ve heard comments like:
+
+- “Instrumenting my code will make it slower”
+- “Activity Object use memory”
+- “If I instrument my code, it will cost me more as I’ll need bigger servers”
+- “OpenTelemetry is immature and isn’t memory efficient”
+- “OpenTelemetry causes performance issues”
+
+I can only assume that these are based on previous versions, or things like
+OpenTracing/OpenCensus (the heritage frameworks that were the feeders for
+OpenTelemetry). This is not true of the current state of OpenTelemetry, and in
+this post, I’ll provide evidence as to why this isn’t the case. My hope is that
+this evidence will stop the debate of performance impact so that teams can
+instead debate the value of instrumenting their code.
+
+TL;DR: if you’re optimizing at the level where OpenTelemetry and Activity usage
+would be a concern, you should probably be more worried about the .NET framework
+as a whole.
+
+## Activity vs. OpenTelemetry vs. OpenTelemetry Shim
+
+The first thing to note is developers just now getting to grips with the
+difference between using the `Activity` class in .NET and using the
+OpenTelemetry SDKs to process and export telemetry to a backend.
+
+Activity is the base of all instrumentation in .NET. It’s not tied to a third
+party, or even a particular OpenSource project like OpenTelemetry. It’s in the
+Base Class Library (BCL), which means it’s baked into .NET. You can instrument
```

Comment:
perhaps a good alternative is to say part of github.com/dotnet/runtime.

Reply:
I think BCL is more well-known than the github repo. This might be what Microsoft are moving towards but I think it confuses the average reader.

Reply:
I'm really on the fence about this one. Calling it the BCL is categorically incorrect, but it's a term that's been burned into the brains of .NET developers for 1.5 decades.

FWIW I don't think it confuses the reader. Maybe a parenthetical `it's in the runtime (BCL)`

Reply:
This post is for the reader to understand, rather than being "technically correct". Above all else, it should be educational, and easily understandable to a large amount of readers.

I can understand that it's not the correct use of the term, however, as Phillip says, this is the way that majority of .NET developers see it.

The ones that understand what runtime means will shrug this off, and the ones that don't will not be mislead to a point that they make a mistake. Whereas the other way will make it less accessible to the people who don't understand what the runtime is.